### ISSUE TITLE
Incorrect behaviour with numeric and currency fields

### ISSUE THREAD

**Issue by boncri** [pods-framework/pods#5651]
## Issue Overview
Incorrect format of numeric and currency fields with italian locale settings.

## Expected Behavior
Setting a numeric or currency field value with decimals and using a format like "1.200,00" (dot as thousand separator and comma as decimal separator), the saved value is incorrect and the decimal separator is ignored.

The problem is not present for formats like "1,200.00" (comma as thousand separator and dot as decimal separator).

The problem wasn't present with the version 2.7.14 of the plugin.

## Current Behavior
Setting a value of 7,50 with 2 decimals leads to a saved value of 750,00.

## Steps to Reproduce (for bugs)
1. Create a numeric or currency field and set in additional options the "1,234.00" local format with 2 decimals.
2. Create a new pod and set the value to a decimal value, like 7,50.
3. The saved value will be 750,00.

## Possible Solution
The problem wasn't present in version 2.7.14 so it seems a regression bug.

## WordPress Environment
<details>
```
WordPress Version: 5.3.2

PHP Version: 7.1.33

MySQL Version: 5.5.5

Server Software: LiteSpeed

Your User Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_3) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/13.0.5 Safari/605.1.15

Session Save Path: /var/cpanel/php/sessions/ea-php71

Session Save Path Exists: Yes

Session Save Path Writeable: Yes

Session Max Lifetime: 1440

Opcode Cache:

Apc: No
Memcached: No
OPcache: Yes
Redis: No
Object Cache:

APC: No
APCu: No
Memcache: No
Memcached: Yes
Redis: No
WPDB Prefix: wp_

WP Multisite Mode: No

WP Memory Limit: 40M

Current Memory Usage: 13,229M

Current Memory Usage (real): 14,000M

Pods Network-Wide Activated: No

Pods Install Location: /home/tnfluzoq/public_html/wp-content/plugins/pods/

Pods Tableless Mode Activated: No

Pods Light Mode Activated: No

Currently Active Theme: NMBS

Currently Active Plugins:

Autoptimize: 2.6.2
Pods - Custom Content Types and Fields: 2.7.17.1
WP Statistics: 12.6.12```
</details>


## Workaround or Alternate Solution Until Bug is Addressed
Using numeric and currency fields with 1,234.00 as locale format.

## Related Issues and/or PRs

**Comment by JoryHogeveen** [comment]
@boncri 

Please test the changes in #5653 and confirm this is working.
I also added tests to our compiler to make sure these type of bugs do not happen again.

**Comment by boncri** [comment]
@JoryHogeveen 

I tested the fix in #5653 and it works perfectly.

Thank you for the quick fix! 👍
